### DISCUSSION TITLE
how to debug the error like "constraint #12364464 is not satisfied:... when call `proof, err := groth16.Prove(r1cs, pk, witness)`

### DISCUSSION THREAD

**Discussion by xiangjianmeng** [Consensys/gnark#1184] [General]
I encountered this error, when execute `proof, err := groth16.Prove(r1cs, pk, witness)`

```
00:30:44 ERR error="constraint #12364464 is not satisfied: 1 ⋅ 3681695053205457254167776065514545950670305446672035081491616726849843773852 != 15211933761486334393069776398049289876332597998649722777068774480811349022785" nbConstraints=12459671
panic: constraint #12364464 is not satisfied: 1 ⋅ 3681695053205457254167776065514545950670305446672035081491616726849843773852 != 15211933761486334393069776398049289876332597998649722777068774480811349022785
```

how to locate which constraint is incorrect in circuit. Circuit could be executed successfully in unit test test.IsSolved.

**Comment by ivokub** [comment]
So, when you run `test.IsSolved(circuit, witness, ecc.BN254.ScalarField())` you don't get an error but when you run `groth16.Prove(r1cs, pk, witness)`, you do? In that case, it may be a bug. Could you post the full example?
